src/Entity/FieldableEdgeEntityBase.php

```diff
@@ -536,4 +538,143 @@ public function getIterator(): \Traversable {
     return new \ArrayIterator($this->getFields());
   }
 
+  /**
+   * The plain data values of the contained fields.
+   *
+   * This always holds the original, unchanged values of the entity. The values
+   * are keyed by language code to follow core's suit, regardless that
+   * Apigee Edge entities are NOT translatable.
+   *
+   * @todo: Add methods for getting original fields and for determining
+   * changes.
+   * @todo: Provide a better way for defining default values.
+   *
+   * @var array
+   */
+  private $values = [];
+
+  /**
+   * Implements the magic method for getting object properties.
+   *
+   * @see https://www.drupal.org/project/drupal/issues/3281720
+   * @see \Drupal\Core\Entity\ContentEntityBase
+   */
+  public function &__get($name) {
+    // If this is an entity field, handle it accordingly. We first check whether
+    // a field object has been already created. If not, we create one.
+    if (isset($this->fields[$name][LanguageInterface::LANGCODE_NOT_SPECIFIED])) {
+      return $this->fields[$name][LanguageInterface::LANGCODE_NOT_SPECIFIED];
+    }
+    // Inline getFieldDefinition() to speed things up.
+    if (!isset($this->fieldDefinitions)) {
+      $this->getFieldDefinitions();
+    }
+    if (isset($this->fieldDefinitions[$name])) {
+      $return = $this->getField($name);
+      return $return;
+    }
+    // Else directly read/write plain values. That way, non-field entity
+    // properties can always be accessed directly.
+    if (!isset($this->values[$name])) {
+      $this->values[$name] = NULL;
+    }
+    return $this->values[$name];
+  }
+
+  /**
+   * Implements the magic method for setting object properties.
+   *
+   * Uses default language always.
+   */
+  public function __set($name, $value) {
+    // Inline getFieldDefinition() to speed things up.
+    if (!isset($this->fieldDefinitions)) {
+      $this->getFieldDefinitions();
+    }
+    // Handle Field API fields.
+    if (isset($this->fieldDefinitions[$name])) {
+      // Support setting values via property objects.
+      if ($value instanceof TypedDataInterface) {
+        $value = $value->getValue();
+      }
+      // If a FieldItemList object already exists, set its value.
+      if (isset($this->fields[$name][LanguageInterface::LANGCODE_NOT_SPECIFIED])) {
+        $this->fields[$name][LanguageInterface::LANGCODE_NOT_SPECIFIED]->setValue($value);
+      }
+      // If not, create one.
+      else {
+        $this->getField($name)->setValue($value);
+      }
+    }
+    // The translations array is unset when cloning the entity object, we just
+    // need to restore it.
+    elseif ($name == 'translations') {
```

Comment:
Translations are not supported by Apigee entities, this should be a dead code.

Reply:
Removed, thanks.